client/signup/navigation-link/index.jsx

```diff
@@ -22,10 +23,36 @@ const NavigationLink = React.createClass( {
 		flowName: React.PropTypes.string.isRequired,
 		positionInFlow: React.PropTypes.number,
 		previousPath: React.PropTypes.string,
-		signupProgressStore: React.PropTypes.array,
+		signupProgress: React.PropTypes.array,
 		stepName: React.PropTypes.string.isRequired
 	},
 
+	/**
+	 * Returns the previous step name, skipping over steps with the
+	 * `wasSkipped` property.
+	 *
+	 * @return {string|null} The previous step name
+	 */
+	getPreviousStepName() {
+		const { stepName, signupProgress } = this.props;
+
+		const currentStepIndex = findIndex( signupProgress, { stepName } );
+
+		let stepIndex = currentStepIndex - 1;
+
+		while ( stepIndex > -1 ) {
```

Comment:
This needs a slight tweak to replicate the current behavior:

```js
const previousStep = find( signupProgress.slice( 0, currentStepIndex ).reverse(), step => ! step.wasSkipped );
return perviousStep ? previousStep.stepName : null;
```

I'll update it.